src/DurableTask.Netherite/OrchestrationService/Client.cs

```diff
@@ -87,7 +87,10 @@ public void Process(ClientEvent clientEvent)
             if (clientEvent is ClientEventFragment fragment)
             {
                 var originalEventString = fragment.OriginalEventId.ToString();
-                var index = (originalEventString, clientEvent.ReceiveChannel);
+
+                var group = fragment.GroupId.HasValue
+                  ? fragment.GroupId.Value.ToString()       // groups are now the way we track fragments
+                  : $"{fragment.OriginalEventId}-{fragment.ReceiveChannel}";  // prior to introducing groups, we used event id and channel, which is not always good enough
```

Comment:
Do you still need `originalEventString` on line 89? I was expecting that you'd reuse that local variable here but maybe it can be removed?

Reply:
Good catch. Yes, it can be removed.

Reply:
Actually, I just realized it is being used by trace statements further down. So I will reuse it.